components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -1560,4 +1572,8 @@ void BraveWalletService::GetAnkrSupportedChainIds(
   std::move(callback).Run(std::move(chain_ids));
 }
 
+void BraveWalletService::IsPrivateWindow(IsPrivateWindowCallback callback) {
+  std::move(callback).Run(is_private_window_);
```

Comment:
We don't need new field `is_private_window_`. Just return `profile_prefs_->IsIncognitoProfile()` here